src/services/videoconference/logic/index.js

```diff
@@ -0,0 +1,102 @@
+const rp = require('request-promise-native');
+const xml2js = require('xml2js-es6-promise');
+const { error } = require('../../../logger');
+const {
+	MESSAGE_KEYS, RETURN_CODES, ROLES, GUEST_POLICIES,
+} = require('./constants');
+
+const createParams = { allowStartStopRecording: false, guestPolicy: GUEST_POLICIES.ALWAYS_DENY };
+
+/**
+ * responseheaders will be evaluated to have a set-cookie valuecontaining a jsessionid
+ * @param {*} headers
+ */
+const getSessionCookieFromHeaders = (headers) => {
+	if (headers && headers['set-cookie'] && Array.isArray(headers['set-cookie'])) {
+		return headers['set-cookie'].find((cookie) => String(cookie).startsWith('JSESSIONID'));
+	}
+	return undefined;
+};
+
+/**
+ * creates a url for attendee or moderator to join a meeting.
+ * if the meeting does not exist, it will be created.
+ *
+ * @returns join url
+ */
+exports.createMeeting = (
+	server, meetingName, meetingId, userName, role, params,
+) => server.administration
+	.create(meetingName, meetingId, createParams)
+	.then((meeting) => {
+		// here we probably have a meeting, add user to the meeting...
+		const { response } = meeting;
+		if (!meeting || !response) {
+			throw new Error('error contacting bbb/server');
+		}
+		if (!Array.isArray(response.returncode) || !response.returncode.includes(RETURN_CODES.SUCCESS)) {
+			const message = 'meeting room creation failed';
+			error(message, response);
+			throw new Error(message);
+		}
+		let secret;
+		switch (role) {
+			case ROLES.MODERATOR:
+				if (!Array.isArray(response.moderatorPW) || !response.moderatorPW.length) {
+					throw new Error('invalid moderator credentials');
+				}
+				secret = response.moderatorPW[0];
+				break;
+			case ROLES.ATTENDEE:
+			default:
+				if (!Array.isArray(response.attendeePW) || !response.attendeePW.length) {
+					throw new Error('invalid attendee credentials');
+				}
+				secret = response.attendeePW[0];
+		}
+
+		if (!Array.isArray(response.meetingID) || !response.meetingID.length) {
+			throw new Error('invalid meetingID');
+		}
+		const p = Object.assign({}, { redirect: false }, params);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use an object spread instead of `Object.assign` eg: `{ ...foo }`.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4819756)